### PR TITLE
Add message type for plugins

### PR DESCRIPTION
**Pull request by MiniDigger**
This adds a new public api to send messages, with their own message type so that its clearly visible to a user where the message came from.
(it also changes some internals to make the package info available in the api)

Usage (in some command for example):
```js
const red = '04';
client.sendMessage(red + "You did something wrong", target.chan);
client.sendMessage("Try this instead", target.chan);
```

Ideally publicClient would automatically know who just called the method so that plugins don't need to pass their identifier every time, but I wasn't sure on how to accomplish that nicely. 
We could have a "shortName" in the package info for that I guess.

### CODE REVIEW

File: client/components/Message.vue [hunk 1]
```diff
@@ -40,6 +40,13 @@
 					<span class="only-copy">&gt; </span>
 				</template>
 			</span>
+			<span v-else-if="message.type === 'plugin'" class="from">
+				<template v-if="message.from && message.from.nick">
+					<span class="only-copy">[</span>
+					*{{ message.from.nick }}
```
Comment (xPaw) [inline client/components/Message.vue]: We can style this nicer, give it an icon with FA for example.

File: src/plugins/packages/publicClient.js [hunk 2]
```diff
@@ -37,4 +39,24 @@ module.exports = class PublicClient {
...
+	 * @param {String} identifier the identifier/name of the packages that is sending this message
+	 */
+	sendMessage(text, chan, identifier) {
+		chan.pushMessage(
+			this.client,
+			new Msg({
+				type: Msg.Type.PLUGIN,
+				text: text,
+				from: {
+					nick: identifier,
```
Comment (xPaw) [inline src/plugins/packages/publicClient.js]: Perhaps we can use (or at least default to) package's name?

**Comment by xPaw** [comment]
Can you make a corresponding PR to the site repo documenting this? Thanks.

**Comment by MiniDigger** [comment]
sure, https://github.com/thelounge/thelounge.github.io/pull/220

**Pull request merged**